operator/controllers/kyma_controller.go

```
@@ -137,29 +140,76 @@ func (r *KymaReconciler) HandleProcessingState(ctx context.Context, logger *logr
 	return nil
 }
 
-func (r *KymaReconciler) HandleDeletingState(_ context.Context) error {
-	return nil
-}
+func (r *KymaReconciler) HandleDeletingState(ctx context.Context, logger *logr.Logger, kyma *operatorv1alpha1.Kyma) (bool, error) {
+	templates, err := release.GetTemplates(r, ctx, kyma)
+	if err != nil {
+		return false, r.KymaStatus().UpdateStatus(ctx, kyma, operatorv1alpha1.KymaStateError,
+			"deletion cannot proceed - templates could not be fetched")
```

Same as above: we should probably propagate the error message for transparency.